lib/doorkeeper/openid_connect.rb

```diff
@@ -26,11 +26,30 @@
 
 module Doorkeeper
   module OpenidConnect
-    # TODO: make this configurable
-    SIGNING_ALGORITHM = 'RS256'.freeze
+    def self.signing_algorithm
+      configuration.signing_algorithm.to_s.upcase.to_sym
+    end
 
     def self.signing_key
-      JSON::JWK.new(OpenSSL::PKey.read(configuration.jws_private_key))
+      key =
+        if [:HS256, :HS384, :HS512].include?(signing_algorithm)
+          configuration.signing_key
+        else
+          OpenSSL::PKey.read(configuration.signing_key)
+        end
+      JSON::JWK.new(key)
+    end
+
+    def self.signing_key_normalized
+      key = signing_key
+      case key[:kty].to_sym
+      when :RSA
+        key.slice(:kty, :kid, :e, :n)
+      when :EC
+        key.slice(:kty, :kid, :x, :y)
+      when :oct
+        key.slice(:kty, :kid, :k)
```

Comment:
Hmm not sure, the [spec has this to say](http://openid.net/specs/openid-connect-core-1_0.html#Signing):

> Symmetric signatures MUST NOT be used by public (non-confidential) Clients because of their inability to keep secrets.

So an `oct` key (which I assume is symmetric) should only be used with trusted clients, and the client does need the `k` value to validate the signature, so I think we need to leave it in and it's up to the user to decide if the client can be trusted?